.circleci/config.yml

```diff
@@ -16,4 +20,31 @@ jobs:
             - ./node_modules
       - run: npm run lint
       - run: npm test
-      # See https://circleci.com/docs/2.0/deployment-integrations/ for deploy examples
+      - persist_to_workspace:
+          root: ~/repo
+          paths: .
+
+  deploy:
+    <<: *common
+    steps:
+      - attach_workspace:
+          at: ~/repo
+      - run: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/repo/.npmrc
+      - run: npm publish
+
+workflows:
+  version: 2
+  build-deploy:
+    jobs:
+      - build:   # run for all branches and PRs
```

Comment:
Do we need to include steps like running our test suite and linter here? I see that's what USWDS is doing. Even though we run these steps before a PR can be merged, it seems like there could potentially be a risk of 2 PRs being merged in such a way that PR#2 could break a test, but CI wouldn't catch it in the PR review stage because it's being reviewed against master, and the test is something introduced by PR#1. Because we don't require rebasing on master before master, this kind of situation, while unlikely, is possible.

Reply:
I have gotten dizzy trying to figure these scenarios out over the past week and make them less error prone. The way I've done it on other repos is to push a tagged commit directly to master. When you do that there's no risk of missing a commit to master because when you try to push, git will stop you since your local doesn't contain the newer commit.

Do you know the folks who are working on USWDS? I'd be interested in knowing a step-by-step on how they do releases because their CircleCI file is simpler.

Reply:
Yes! I can introduce you to some folks at USWDS. I'll send an email introduction.